pages/questionnaireCrowd.js

```diff
@@ -0,0 +1,452 @@
+} from '../components/Questionnaire/enhancers'
+import questionnaire, { description } from './questionnaire'
+import { withRouter } from 'next/router'
+import QuestionnaireActions from '../components/Questionnaire/QuestionnaireActions'
+import Frame from '../components/Frame'
+import Questionnaire from '../components/Questionnaire/Questionnaire'
+import {
+  userDetailsFragment,
+  withMyDetails
+} from '../components/Account/enhancers'
+import { errorToString } from '../lib/utils/errors'
+import {
+  COUNTRIES,
+  fields as addressFields
+} from '../components/Account/AddressForm'
+import FieldSet from '../components/FieldSet'
+import gql from 'graphql-tag'
+import DetailsForm from '../components/Account/DetailsForm'
+import { Interaction, RawHtml, colors } from '@project-r/styleguide'
+import { css } from 'glamor'
+import MdArrow from 'react-icons/lib/md/trending-flat'
+
+const { Headline, P } = Interaction
+
+const mutation = gql`
+  mutation submitQuestionnaireAndUpdateMe(
+    $questionnaireId: ID!
+    $phoneNumber: String
+    $address: AddressInput
+  ) {
+    updateMe(phoneNumber: $phoneNumber, address: $address) {
+      id
+      ...Details
+    }
+    submitQuestionnaire(id: $questionnaireId) {
+      id
+      userSubmitDate
+      userHasSubmitted
+    }
+  }
+  ${userDetailsFragment}
+`
+
+const withMutation = graphql(mutation, {
+  props: ({ mutate }) => ({
+    submitForm: variables =>
+      mutate({
+        variables
+      })
+  })
+})
+
+const meta = {
+  title: 'Crowd Umfrage',
+  description: t('questionnaire/description'),
+  facebookTitle: t('pages/meta/questionnaire/socialTitle'),
+  facebookDescription: t('pages/meta/questionnaire/socialDescription'),
+  twitterTitle: t('pages/meta/questionnaire/socialTitle'),
+  twitterDescription: t('pages/meta/questionnaire/socialDescription'),
+  facebookImage: `${CDN_FRONTEND_BASE_URL}/static/social-media/umfrage/2018/facebookImage.png`,
+  twitterImage: `${CDN_FRONTEND_BASE_URL}/static/social-media/umfrage/2018/twitterImage.png`
+}
+
+const DEFAULT_COUNTRY = COUNTRIES[0]
+
+const styles = {
+  intro: css({
+    margin: '35px 0 70px'
+  }),
+  arrow: css({
+    marginRight: 15,
+    minWidth: 16,
+    color: colors.lightText
+  }),
+  thankYouItem: css({
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'left',
+    alignItems: 'center',
+    marginBottom: 20
+  })
+}
+
+const getValues = me => {
+  let addressState = {}
+  if (me.address) {
+    addressState = {
+      name: me.address.name || me.name,
+      line1: me.address.line1,
+      line2: me.address.line2,
+      postalCode: me.address.postalCode,
+      city: me.address.city,
+      country: me.address.country
+    }
+  } else if (me) {
+    addressState.name = [me.firstName, me.lastName].filter(Boolean).join(' ')
+    addressState.country = DEFAULT_COUNTRY
+  }
+
+  return {
+    phoneNumber: me.phoneNumber || '',
+    ...addressState
+  }
+}
+
+const isEmptyAddress = (values, me) => {
```

Comment:
This could probably be moved to `AddressForm` and exported from there to here and `UpdateMe`.